src/visit.rs

```diff
@@ -367,7 +404,23 @@ impl<'ast> Visit<'ast> for DiscoveryVisitor<'_> {
             trace!("mod excluded by attrs");
             return;
         }
-        self.mod_namespace_stack.push(mod_name.clone());
+
+        // Extract path attribute value, if any (e.g. `#[path="..."]`)
+        let path_attribute = match find_path_attribute(&node.attrs) {
+            Ok(path) => path,
+            Err(path_attribute) => {
+                error!(?path_attribute, %mod_name, "invalid filesystem traversal in mod path attribute");
+                self.failed = true;
+                return;
+            }
+        };
+        let mod_namespace = ModNamespace {
+            name: mod_name,
+            path_attribute,
+            source_location: node.mod_token.span().start().into(),
```

Comment:
I think this will highlight specifically the `mod` token but we probably want to save the span of the `node`, i.e. the whole statement. It's not a huge deal because the keyword will obviously be close and this is relatively unlikely to be hit, but it probably sets a better example for adding more error location info in future.

Reply:
Good point, updated to span of the whole node